app/templates/carpools/show.html

```diff
@@ -77,6 +77,11 @@ <h4>Carpool Status</h4>
                         <a class="btn btn-primary" href="{{ url_for('carpool.edit', uuid=pool.uuid) }}">Edit your carpool</a>
                     {% else %}
                         <a class="btn btn-danger" href="{{ url_for('admin.admin_cancel_carpool', uuid=pool.uuid) }}">Cancel this carpool</a>
+                        {%- if pool.seats_available > 0 %}
+                            <a class="btn btn-primary" href="{{ url_for('carpool.new_rider', carpool_uuid=pool.uuid) }}">Request a seat in carpool</a>
```

Comment:
Seems a strange place for this logic (showing the 'request a ride' UI deep in code that's for drivers). Why not have a wholly separate branch for current_user.has_roles('admin') instead of tucking it into the driver branch?

Reply:
I agree, and that was my first idea... however this requires several deep changes in the code as the logic is a bit convoluted right now. The branches are now more like: 
- if current_user_ride_request and current_user_ride_request.status == 'approved'
- (else) current_user_ride_request
- (else) driver or admin
- (else) everybody else

This change was quicker to solve the bug, that is a showstopper for creating local test data and bug solving. I was thinking about solving the bug and then adding an 'Enhancement" issue to clean the code

Reply:
Better to fix it properly the first time.

Reply:
Changed. Kept the branches but relocated the admin cancel code added in a previous version that was creating trouble. A separate admin branch would add more complexity as an admin can be any of those three types.